rust/gateway/src/eventloop.rs

```diff
@@ -251,33 +283,36 @@ impl Eventloop {
         }
     }
 
-    pub fn allow_access(&mut self, result: Result<Vec<IpNetwork>, Timeout>, req: AllowAccess) {
+    pub fn allow_access(&mut self, result: Result<Vec<IpAddr>, Timeout>, req: AllowAccess) {
         let addresses = result
             .inspect_err(|e| tracing::debug!(client = %req.client_id, reference = %req.reference, "DNS resolution timed out as part of allow access request: {e}"))
             .unwrap_or_default();
```

Comment:
I think this `unwrap_or_default` will also not play well with your `is_empty` check inside `allow_access`.